1. 编写Python程序，打印"数据科学与工程导论"，并使用 print(chr(0x2605)) 语句将星星包围起来。

In [1]:
print(chr(0x2605) + " 数据科学与工程导论 " + chr(0x2605))

★ 数据科学与工程导论 ★


    ★ 数据科学与工程导论 ★

2. 写一个 Python 程序，输入 x, y, z 这三个数，将这三个数从小到大使用 print 函数打印出来。

In [ ]:
x = int(input("请输入x: "))
y = int(input("请输入y: "))
z = int(input("请输入z: "))

numbers = [x, y, z]

print(sorted(numbers))

    Input: 6 9 5
    Output: 5 6 9

3. 写一个 Python 程序，有 w, x, y, z 四个数，将这四个数从大到小使用 print 函数打印出来。

In [2]:
x = 10
y = 20
z = 30
w = 1

numbers = [x, y, z, w]

print(sorted(numbers, reverse=True))

[30, 20, 10, 1]


    Output: [30, 20, 10, 1]

4. 写一个 Python 程序，输出 1-100 中的所有奇数。

In [3]:
for i in range(1, 100):
    if i % 2 != 0:
        print(i )

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49
51
53
55
57
59
61
63
65
67
69
71
73
75
77
79
81
83
85
87
89
91
93
95
97
99


    Output:1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49
51
53
55
57
59
61
63
65
67
69
71
73
75
77
79
81
83
85
87
89
91
93
95
97
99


5. 写一个 Python 程序，用 for 循环，求解 1 到 100 的和。

In [4]:
total_sum = 0
for i in range(1, 101):
    total_sum += i

print(total_sum)

5050


    Output:5050

6. 写一个 Python 程序，分别用 for 和 while 循环实现对一个给定序列的倒排序输出。例如，给定 L=[1,2,3,4,5]，输出为 [5,4,3,2,1]。

In [5]:
L = [1, 2, 3, 4, 5]

reversed_L_for = []
for i in range(len(L)-1, -1, -1):
    reversed_L_for.append(L[i])
print(reversed_L_for)

reversed_L_while = []
i = len(L) - 1
while i >= 0:
    reversed_L_while.append(L[i])
    i -= 1
print(reversed_L_while)

[5, 4, 3, 2, 1]
[5, 4, 3, 2, 1]


    Output:[5, 4, 3, 2, 1]
    [5, 4, 3, 2, 1]

7. 写一个 Python 程序，判断一个输入的字符串 S 是否包含由两个或两个以上连续出现的相同字符组成的字符串。例如，abccccda 中就包含 cccc 这个由 4 个连续字符组成的字符串。

In [ ]:
def if_continue_2chars(S):
    for i in range(len(S) - 1):
        if S[i] == S[i + 1]:
            return True
    return False

S = input("Input S: ")
if if_continue_2chars(S):
    print("Yes")
else:
    print("No")

    Input:dahsidhasaaa
    Output:Yes

8. 写一个 Python 程序，输入一个字符串 S，去掉其中所有的空格后输出。例如，“DataScience and Engineering”，去掉空格后为“DataScienceandEngineering”。

In [ ]:
S = input("Input S: ")
S_no_spaces = S.replace(" ", "")
print(S_no_spaces)

    Input:DataScience and Engineering
    Output:DataScienceandEngineering

9. 请设计一个求 3 次方根的算法（不允许直接调用求方根的函数），并给出对应的 Python 程序。
采用牛顿迭代法求解，迭代过程数学公式表示为：
$ x_{n+1}=x_n-\frac{f(x_n)}{f^{\prime}(x_n)}$

In [ ]:
def newton_cbrt(n, tolerance=1e-7, max_iterations=1000):
    x = n
    for _ in range(max_iterations):
        next_x = x - (x**3 - n) / (3 * x**2)
        if abs(next_x - x) < tolerance:
            return next_x
        x = next_x
    return x

n = float(input("Input: "))
cbrt_n = newton_cbrt(n)
print(cbrt_n)

    Input:1024
    Output:10.079368399158986

10. 写一个 Python 程序，给定一个常数 n（n>0），求 n 的阶乘，即 n!=1×2×...×(n-1)×n。例如，4!=24，5!=120。

In [ ]:
def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n - 1)

n = int(input("Input: "))
if n > 0:
    print(f"{n}! = {factorial(n)}")
else:
    print("Error")

    Input:4!
    Output:24